Sources/Compiler/AST/AST+Invariants.swift

```diff
@@ -0,0 +1,209 @@
+import Utils
+
+extension AST {
+
+  /// Returns `.success` if `decl` is a well-formed top-level or namespace member declaration.
+  /// Otherwise, returns `.failure` with the diagnostics of the broken invariants.
+  func isValidGlobalScopeMember(
+    _ decl: AnyDeclID,
+    atTopLevel: Bool
+  ) -> SuccessOrDiagnostics {
+    var ds: [Diagnostic] = []
```

Comment:
The name `ds` is a weird grammatical mix. I read it as “Ds,” i.e. a collection of D objects, whatever D is supposed to mean.  What is it about being a collection that warrants giving it special status and injecting type information into its name?

```suggestion
    var report: [Diagnostic] = []
```

or even 

```suggestion
    var log: [Diagnostic] = []
```

Reply:
OK, reasonable request. I'll create an issue.